app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -46,6 +48,19 @@ def create_sub_plans
         end
       end
 
+      def spawn_plans
+        super
+        if output[:planned_count] - output[:remote_triggered_count] >= proxy_batch_size
+          plan_event(Actions::TriggerProxyBatch::TriggerNextBatch, nil, step_id: input[:trigger_run_step_id])
+          output[:remote_triggered_count] += proxy_batch_size
+        end
+      end
```

Comment:
`spawn_plans` in one of the ancestors suspends the action, nothing after this `super` will be executed. 

```suggestion
      def spawn_plans
        super
      ensure
        if output[:planned_count] - output[:remote_triggered_count] >= proxy_batch_size
          plan_event(Actions::TriggerProxyBatch::TriggerNextBatch, nil, step_id: input[:trigger_run_step_id])
          output[:remote_triggered_count] += proxy_batch_size
        end
      end
```